File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.dart
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *   int val;
 *   TreeNode? left;
 *   TreeNode? right;
 *   TreeNode([this.val = 0, this.left, this.right]);
 * }
 */
class Solution {
    List<int> sortedList = [];

    void postOrder(TreeNode? root)
    {
        if(root != null) {
            postOrder(root.left);
            postOrder(root.right);
            sortedList.add(root.val);
        }
    }
  List<int> postorderTraversal(TreeNode? root) {
   
    postOrder(root);
    return sortedList;
  }
}

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> res = new ArrayList<>();

    void postOrder(TreeNode root)
    {
        if(root != null) {
            postOrder(root.left);
            postOrder(root.right);
            res.add(root.val);
        }
    }
    public List<Integer> postorderTraversal(TreeNode root) {
        postOrder(root);
        return res;
    }
}